src/middleware/validate.ts

```diff
@@ -0,0 +1,46 @@
+import { MiddlewareHandler } from "hono";
+import { ServiceType, Error } from "../types";
+import { toEnvKey } from "../utils";
+import { AppContext } from "../bindings";
+
+function validate(): MiddlewareHandler {
+  return async (context: AppContext, next) => {
+    const headers = context.req.raw.headers;
+    const xGatewayServiceHost = headers.get("x-gateway-service-host");
+    const xGatewayServiceType = (headers.get("x-gateway-service-type") as ServiceType) || "DIRECT";
```

Comment:
Insead of using "DIRECT" string can we use the enum?

Reply:
It is already a type-union, so the fallback cannot be misspelled. Your IDE will show an error if you even attempt to write it in lowercase.

Reply:
I have converted it into a class to have a parse function for more maintainability.

Reply:
Nope, no issue. just different than Swift.